app/services/resolve_account_service.rb

```diff
@@ -86,10 +89,12 @@ def process_webfinger!(uri, redirected = false)
     elsif !redirected
       return process_webfinger!("#{confirmed_username}@#{confirmed_domain}", true)
     else
-      raise WebfingerRedirectError, "The URI #{uri} tries to hijack #{@username}@#{@domain}"
+      raise Webfinger::RedirectError, "The URI #{uri} tries to hijack #{@username}@#{@domain}"
     end
 
     @domain = nil if TagManager.instance.local_domain?(@domain)
+  rescue Webfinger::GoneError
+    @gone = true
```

Comment:
Note that we currently return `410 Gone` in webfinger for suspended accounts, regardless of whether the suspension is final: https://github.com/tootsuite/mastodon/blob/acc1c038610a82d8470acde6753716da0dc4b1fa/app/controllers/well_known/webfinger_controller.rb#L37-L39

It should at the very least be changed for this PR.